src/components/IDO/IDO.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useEffect } from 'react'
+import { Grid, Typography, Box, CardMedia } from '@material-ui/core'
+import { PublicKey } from '@solana/web3.js'
+import { Decimal, PoolStructure } from '@invariant-labs/sdk/lib/market'
+import { BN } from '@project-serum/anchor'
+import useStyles from './style'
+import ExchangeAmountInput from '@components/Inputs/ExchangeAmountInput/ExchangeAmountInput'
+import { printBN, printBNtoBN } from '@consts/utils'
+import { OutlinedButton } from '@components/OutlinedButton/OutlinedButton'
+import watchIcon from '@static/svg/watch.svg'
+import solanaIcon from '@static/svg/solanaToken.svg'
+import dollarIcon from '@static/svg/dollarToken.svg'
+import invariantIcon from '@static/svg/invariantToken.svg'
+import invariantLogo from '@static/svg/invariantLogo.svg'
+import { SwapToken } from '@components/Swap/Swap'
+import AnimatedNumber from '@components/AnimatedNumber'
+import { Status } from '@reducers/solanaWallet'
+import { fromFee } from '@invariant-labs/sdk/lib/utils'
+
+export interface IIDO {
+  tokens: SwapToken[]
+  totalDeposit: number
+  totalSolContribute: number
+  priceOfToken: number
+  valueOfInvariantTokens: number
+  walletStatus: Status
+  claimable: Boolean
+  withdrawable: Boolean
+  currencyInfo: ICurrencyData
+  saleEnd: ITime
+  graceEnd: ITime
+  pools: PoolStructure[]
```

Comment:
Why are you copying stuff from swap? Pools, slippage tolerance, etc., won't be used in IDO, there will be only depositing, withdrawing or claiming some amount of one token. If you don't understand something about the task, don't be afraid to ask some questions.

Reply:
I improved this component, it should work better now :)